### PR TITLE
Add run for tomte workflow

### PR DESCRIPTION
**Pull request by fevac**
### Added

- Run subcommand for tomte workflow

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b tomte_run -a
    ```

### How to test

- [x] dry run resume `cg workflow tomte run -d concisewren` - check that command fails to resume as it cannot find a previous NF-tower id. 
image

- [x] dry run from start `cg workflow tomte run --from-start -d concisewren` - check that launch command is correct
image

- [x] run `cg workflow tomte run --from-start concisewren` - check that case successfully start in tower
image

- [x] run `cg workflow rnafusion run -from-start tendergoose` - check that other workflows still work
image

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

**Comment by fevac** [comment]
tests are passing

File: tests/cli/workflow/nf_analysis/test_cli_run.py [hunk 1]
```diff
@@ -3,163 +3,169 @@
...
 @pytest.mark.parametrize(
-    "context", ["raredisease_context", "rnafusion_context", "taxprofiler_context"]
+    "workflow",
+    [Workflow.RAREDISEASE, Workflow.RNAFUSION, Workflow.TAXPROFILER, Workflow.TOMTE],
 )
-def test_without_samples(
+def test_run_case_without_samples(
     cli_runner: CliRunner,
-    context: CGConfig,
+    workflow: Workflow,
```
Comment (diitaz93) [inline tests/cli/workflow/nf_analysis/test_cli_run.py:70]:
```suggestion
    workflow: str,
```

File: tests/cli/workflow/nf_analysis/test_cli_run.py [hunk 2]
```diff
@@ -3,163 +3,169 @@
...
-        ("taxprofiler_context", "taxprofiler_case_id"),
-    ],
+    "workflow",
+    [Workflow.RNAFUSION, Workflow.TAXPROFILER, Workflow.RAREDISEASE, Workflow.TOMTE],
 )
-def test_without_config(
+def test_run_case_without_config_files(
     cli_runner: CliRunner,
-    context: CGConfig,
+    workflow: Workflow,
```
Comment (diitaz93) [inline tests/cli/workflow/nf_analysis/test_cli_run.py:100]:
```suggestion
    workflow: str,
```

File: tests/cli/workflow/nf_analysis/test_cli_run.py [hunk 3]
```diff
@@ -3,163 +3,169 @@
...
+
+    # THEN command should not exit successfully
     assert result.exit_code != EXIT_SUCCESS
-    # THEN warning should be printed that no config file is found
+
+    # THEN the error log should indicate that the config file is missing
     assert "No config file found" in caplog.text
 
 
+# @pytest.mark.parametrize(
```
Comment (diitaz93) [inline tests/cli/workflow/nf_analysis/test_cli_run.py:121]: Delete this maybe?

File: tests/cli/workflow/nf_analysis/test_cli_run.py [hunk 4]
```diff
@@ -3,163 +3,169 @@
...
-        ("taxprofiler_context", "taxprofiler_case_id"),
-    ],
+    "workflow",
+    [Workflow.RAREDISEASE, Workflow.RNAFUSION, Workflow.TAXPROFILER, Workflow.TOMTE],
 )
-def test_with_config(
+def test_run_case_from_start_dry_run(
     cli_runner: CliRunner,
-    context: CGConfig,
+    workflow: Workflow,
```
Comment (diitaz93) [inline tests/cli/workflow/nf_analysis/test_cli_run.py:151]:
```suggestion
    workflow: str,
```

File: tests/cli/workflow/nf_analysis/test_cli_run.py [hunk 5]
```diff
@@ -171,70 +177,66 @@ def test_with_config(
...
-        ("taxprofiler_context", "taxprofiler_case_id"),
-    ],
+    "workflow",
+    [Workflow.RNAFUSION, Workflow.TAXPROFILER, Workflow.RAREDISEASE, Workflow.TOMTE],
 )
-def test_with_revision(
+def test_run_case_with_revision_dry_run(
     cli_runner: CliRunner,
-    context: CGConfig,
+    workflow: Workflow,
```
Comment (diitaz93) [inline tests/cli/workflow/nf_analysis/test_cli_run.py:185]:
```suggestion
    workflow: str,
```

File: tests/cli/workflow/nf_analysis/test_cli_run.py [hunk 6]
```diff
@@ -171,70 +177,66 @@ def test_with_config(
...
-        ("taxprofiler_context", "taxprofiler_case_id"),
-    ],
+    "workflow",
+    [Workflow.RNAFUSION, Workflow.TAXPROFILER, Workflow.RAREDISEASE, Workflow.TOMTE],
 )
-def test_resume_with_id(
+def test_resume_case_dry_run(
     cli_runner: CliRunner,
-    context: CGConfig,
+    workflow: Workflow,
```
Comment (diitaz93) [inline tests/cli/workflow/nf_analysis/test_cli_run.py:219]:
```suggestion
    workflow: str,
```

File: tests/cli/workflow/nf_analysis/test_cli_run.py [hunk 7]
```diff
@@ -245,65 +247,57 @@ def test_resume_with_id(
...
-        ("taxprofiler_context", "taxprofiler_case_id"),
-    ],
+    "workflow",
+    [Workflow.RNAFUSION, Workflow.TAXPROFILER, Workflow.RAREDISEASE, Workflow.TOMTE],
 )
-def test_resume_without_id_error(
+def test_resume_case_with_missing_tower_id(
     cli_runner: CliRunner,
-    context: CGConfig,
+    workflow: Workflow,
```
Comment (diitaz93) [inline tests/cli/workflow/nf_analysis/test_cli_run.py:255]:
```suggestion
    workflow: str,
```

File: tests/cli/workflow/nf_analysis/test_cli_run.py [hunk 8]
```diff
@@ -245,65 +247,57 @@ def test_resume_with_id(
...
-        ("taxprofiler_context", "taxprofiler_case_id"),
-    ],
+    "workflow",
+    [Workflow.RNAFUSION, Workflow.TAXPROFILER, Workflow.RAREDISEASE, Workflow.TOMTE],
 )
-def test_with_config_use_nextflow(
+def test_resume_using_nextflow_dry_run(
     cli_runner: CliRunner,
-    context: CGConfig,
+    workflow: Workflow,
```
Comment (diitaz93) [inline tests/cli/workflow/nf_analysis/test_cli_run.py:283]:
```suggestion
    workflow: str,
```

**Review by diitaz93 — APPROVED** [review]
👍

File: tests/cli/workflow/nf_analysis/test_cli_run.py [hunk 9]
```diff
@@ -3,163 +3,169 @@
...
 @pytest.mark.parametrize(
-    "context", ["raredisease_context", "rnafusion_context", "taxprofiler_context"]
+    "workflow",
+    [Workflow.RAREDISEASE, Workflow.RNAFUSION, Workflow.TAXPROFILER, Workflow.TOMTE],
 )
-def test_with_missing_case(
+def test_run_with_missing_case(
     cli_runner: CliRunner,
-    context: CGConfig,
+    workflow: Workflow,
```
[the remark this answers is not in the document]
Comment (fevac) [reply tests/cli/workflow/nf_analysis/test_cli_run.py:41]: isn't it best to leave Workflow? as the only options allowed are the ones in Workflow?

File: tests/cli/workflow/nf_analysis/test_cli_run.py (hunk 3 above)
Comment (fevac) [reply tests/cli/workflow/nf_analysis/test_cli_run.py:121]: yes! I forgot! thanks

File: tests/cli/workflow/nf_analysis/test_cli_run.py (hunk 9 above)
[the remark this answers is not in the document]
Comment (fevac) [reply tests/cli/workflow/nf_analysis/test_cli_run.py:41]: will leave Workflow as it is more specific and clear

**Comment by fevac** [comment]
- deployed

image

image

**Pull request merged**